pkg/kube/multicluster/secretcontroller.go

```diff
@@ -193,6 +200,49 @@ func (c *Controller) Run(stopCh <-chan struct{}) error {
 	return nil
 }
 
+func (c *Controller) onClusterSyncStatusChange(clusterID cluster.ID, status string) {
+	// Only record metrics for clusters we still manage.
+	// This avoids resurrecting metrics for clusters that have been deleted.
+	if !c.cs.Contains(clusterID) {
+		return
+	}
+	c.recordClusterSyncState(clusterID, status)
+}
+
+func (c *Controller) recordClusterSyncState(clusterID cluster.ID, status string) {
+	for _, s := range []string{
```

Comment:
WDYT about having metric like remote_cluster_sync_stats{cluser=remote}: 1,2,3,4 (synced/syncing/timeout/closed) so that we do not 4 labels for every cluster?